### PR TITLE
Update gitignore.plugin: comma separated arguments

### PR DESCRIPTION
**Pull request by low-ghost**
Seems like this is more like the intended effect, splits arguments with commas for proper gitignore.io api request. Thanks for teaching me about custom compdef completions!

### CODE REVIEW

File: plugins/gitignore/gitignore.plugin.zsh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function gi() { curl -sL https://www.gitignore.io/api/$@ ;}
+function gi() { curl -sL https://www.gitignore.io/api/`echo $@ | sed s/" "/,/g` ;}
```
Comment (mcornella) [inline plugins/gitignore/gitignore.plugin.zsh]: A better way to join arguments with a delimiter is with zsh builtin join syntax: `${(j:delimiter:)variable}`. In this case:
```zsh
curl -sL https://www.gitignore.io/api/${(j:,:)@}
```

Comment (low-ghost) [inline plugins/gitignore/gitignore.plugin.zsh]: Whoa killer stuff.Not too familiar with zsh specific syntax and wouldn't have even thought of something that awesome

Comment (apjanke) [inline plugins/gitignore/gitignore.plugin.zsh]: Check out the [14.3. Parameter Expansion](http://zsh.sourceforge.net/Doc/Release/Expansion.html#Parameter-Expansion) section in the ZSH reference manual if you want to learn more about it. It'll show you tons of stuff you can do like this, possibly replacing a bunch of external calls to `grep`, `sed`, `tr`, and the like.

**Comment by mcornella** [comment]
I merged the short version

**Comment by low-ghost** [comment]
thanks @mcornella

**Pull request closed without merging**